sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetWriteSupport.scala

```diff
@@ -482,7 +482,7 @@ object ParquetWriteSupport {
   val SPARK_ROW_SCHEMA: String = "org.apache.spark.sql.parquet.row.attributes"
 
   def setSchema(schema: StructType, configuration: Configuration): Unit = {
-    schema.map(_.name).foreach(ParquetSchemaConverter.checkFieldName)
+    ParquetSchemaConverter.checkFieldNames(schema)
```

Comment:
The change looks good. I just have a quick question: I saw your other PR refactor the code to unify the schema check for avro/parquet/orc in DDL. Should this setSchema and inner field check code be refactored and unified too for avro and ocr?

Reply:
> The change looks good. I just have a quick question: I saw your other PR refactor the code to unify the schema check for avro/parquet/orc in DDL. Should this setSchema and inner field check code be refactored and unified too for avro and ocr?

Will do this in other pr for all v1 insert behavior.